scripts/database.py

```diff
@@ -0,0 +1,182 @@
+import os
+import pickle
+import random
+import re
+
+from tinydb import TinyDB, Query
+
+# save/load livestream related data
+
+SAVEDATA_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'savedata')
+if not os.path.exists(SAVEDATA_PATH):
+    os.mkdir(SAVEDATA_PATH)
+
+
+def save_pickle(savedata):
+    pickle_path = os.path.join(SAVEDATA_PATH, 'livestream_save.pickle')
+    with open(pickle_path, 'wb') as file_handle:
+        pickle.dump(savedata, file_handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+
+def load_pickle(default):
+    pickle_path = os.path.join(SAVEDATA_PATH, 'livestream_save.pickle')
+    if not os.path.exists(pickle_path):
+        return default
+    else:
+        with open(pickle_path, 'rb') as file_handle:
+            return pickle.load(file_handle)
+
+
+# our tinydb
+THE_DB = TinyDB(os.path.join(SAVEDATA_PATH, 'db.json'))
+
+# for url checking
+# slightly modified from https://stackoverflow.com/a/7160778
+url_regex = re.compile(r'^(?:http)s?://'  # http:// or https://
+                       r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|'  # domain...
+                       r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'  # ...or ip
+                       r'(?::\d+)?'  # optional port
+                       r'(?:/?|[/?]\S+)$', re.IGNORECASE)
+
+
+def validate_url(url):
```

Comment:
since whatever gets inputted here gets thrown inside the img source tag for the dj picture element i didn't want broken images but you're right that this doesn't guarantee that